ksqldb-engine/src/main/java/io/confluent/ksql/internal/TelemetryMetricsReporter.java

```diff
@@ -0,0 +1,141 @@
+  public static final String RESOURCE_LABEL_COMMIT_ID = RESOURCE_LABEL_PREFIX + "commit.id";
+  public static final String RESOURCE_CLUSTER_ID = RESOURCE_LABEL_PREFIX + "cluster.id";
+
+  private final Logger logger = LoggerFactory.getLogger(TelemetryMetricsReporter.class);
+  private final TelemetryReporter reporter;
+  private final Map<String, String> metricTags = new HashMap<>();
+  // ToDo: How is the cluster ID passed to this class
+  private final String clusterId = "";
+  private final String metricGroup;
+
+
+  public TelemetryMetricsReporter(
+          final String metricGroup,
+          final String providerNamespace,
+          final String resourceType,
+          final Map<String, ?> config
+  ) {
+    this(metricGroup, providerNamespace, resourceType, config, new TelemetryReporter());
+  }
+
+  protected TelemetryMetricsReporter(
+          final String metricGroup,
+          final String providerNamespace,
+          final String resourceType,
+          final Map<String, ?> config,
+          final TelemetryReporter reporter
+  ) {
+    this.metricGroup = metricGroup;
+    this.reporter = reporter;
+    configureTelemetryReporter(providerNamespace, resourceType, config, reporter);
+  }
+
+  @Override
+  public void report(final List<DataPoint> dataPoints) {
+    for (final DataPoint dataPoint : dataPoints) {
+      emitMetric(
+              dataPoint.getName(),
+              dataPoint.getTime(),
+              dataPoint.getValue(),
+              ImmutableMap.<String, String>builder()
+                      .putAll(metricTags).putAll(dataPoint.getTags()).build()
+      );
+    }
+  }
+
+  void emitMetric(
+          final String metricName,
+          final Instant timestamp,
+          final Object value,
+          final Map<String, String> metricTags
+  ) {
+    // Todo: Is it correct to use the KsqlProvider provided by ce-kafka?
+    final String fullMetricName = MetricsUtils.fullMetricName(
+            KsqlProvider.DOMAIN, metricGroup, metricName);
+    final MetricKey key = new MetricKey(fullMetricName, metricTags);
+
+    reporter.emitter().emitMetric(SinglePointMetric.gauge(key, (Double) value, timestamp));
+  }
+
+  private void configureTelemetryReporter(
+          final String providerNamespace,
+          final String resourceType,
+          final Map<String, ?> config,
+          final TelemetryReporter reporter
+  ) {
+    // Todo: Is it correct to use the KsqlProvider provided by ce-kafka?
+    ProviderRegistry.registerProvider(
+            providerNamespace,
+            KsqlProvider.class.getCanonicalName()
+    );
+
+    reporter.configure(config);
+    final MetricsContext ctx = createMetricContext(providerNamespace, resourceType);
+    try {
+      reporter.contextChange(ctx);
+    } catch (final Throwable e) {
+      logger.error("Failed to properly start up telemetry reporter with the following error: ", e);
+      throw e;
```

Comment:
Do we definitely want to throw here? Just wondering if this would bubble up and cause the server to not start up - if so maybe we don't want to actually throw if the only side effect is not having metrics

Reply:
I think it's ok to throw. I'd rather have us fail to start up than run in some undefined state where we're not emitting metrics. Anyway it's the type of thing that will almost always either work for all clusters running a given release or fail for all clusters.